### ISSUE TITLE
Coverage Issue - await using / IAsyncDisposable

### ISSUE THREAD

**Issue by g0dm0d3** [coverlet-coverage/coverlet#1163]
**Code**:

```
namespace CoverletAsync
{
    public class MyService
    {
        public async Task<byte[]> GetBytes(Func<Task<MemoryStream>> getStream)
        {
            await using var stream = await getStream();
            return stream.ToArray();
        }
    }
}
```

**Test**:

```
namespace CoverletAsync.Tests
{
    public class Tests
    {
        [Test]
        public async Task GetBytes()
        {
            var actual = await new MyService().GetBytes(GetStream);
            Assert.That(actual, Is.EqualTo(new byte[] { 0x01, 0x02 }));
        }

        private static async Task<MemoryStream> GetStream()
        {
            await Task.Delay(1);
            return new MemoryStream(new byte[] { 0x01, 0x02 });
        }
    }
}
```

**Expected**:
Full branch coverage

**Actual**:
Partially covered (1 visits, 4 of 8 branches are covered)

**Output**:
![myservice_coverage](https://user-images.githubusercontent.com/38039872/118234444-ffd4d180-b49b-11eb-9a0e-33d3b081029f.png)

**Report**:

```
<?xml version="1.0" encoding="utf-8"?>
<coverage line-rate="1" branch-rate="0.5" version="1.9" timestamp="1620976353" lines-covered="4" lines-valid="4" branches-covered="4" branches-valid="8">
  <sources>
    <source>F:\</source>
  </sources>
  <packages>
    <package name="CoverletAsync" line-rate="1" branch-rate="0.5" complexity="8">
      <classes>
        <class name="CoverletAsync.MyService/&lt;GetBytes&gt;d__0" filename="Work\experiments\CoverletAsync\CoverletAsync\MyService.cs" line-rate="1" branch-rate="0.5" complexity="8">
          <methods>
            <method name="MoveNext" signature="()" line-rate="1" branch-rate="0.5" complexity="8">
              <lines>
                <line number="10" hits="1" branch="False" />
                <line number="11" hits="1" branch="False" />
                <line number="12" hits="1" branch="True" condition-coverage="50% (4/8)">
                  <conditions>
                    <condition number="211" type="jump" coverage="100%" />
                    <condition number="331" type="jump" coverage="50%" />
                    <condition number="344" type="jump" coverage="0%" />
                    <condition number="373" type="jump" coverage="50%" />
                  </conditions>
                </line>
                <line number="13" hits="1" branch="False" />
              </lines>
            </method>
          </methods>
          <lines>
            <line number="10" hits="1" branch="False" />
            <line number="11" hits="1" branch="False" />
            <line number="12" hits="1" branch="True" condition-coverage="50% (4/8)">
              <conditions>
                <condition number="211" type="jump" coverage="100%" />
                <condition number="331" type="jump" coverage="50%" />
                <condition number="344" type="jump" coverage="0%" />
                <condition number="373" type="jump" coverage="50%" />
              </conditions>
            </line>
            <line number="13" hits="1" branch="False" />
          </lines>
        </class>
      </classes>
    </package>
  </packages>
</coverage>
```

Run: dotnet test /p:CollectCoverage=true /p:CoverletOutputFormat=cobertura

Reproduced in .NET Core 3.1, Coverlet 3.0.3

**Comment by Malivil** [comment]
This might be the same as #914 which is already fixed in the nightlies. Try using a nightly (https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/ConsumeNightlyBuild.md) build and see if it's resolved.

**Comment by daveMueller** [comment]
I can confirm @Malivil answere. It is fixed in the current master branch.

![image](https://user-images.githubusercontent.com/19378678/118894964-81e25180-b905-11eb-9dae-593e5457c279.png)

```
<methods>
   <method name="MoveNext" signature="()" line-rate="1" branch-rate="1" complexity="1">
      <lines>
         <line number="10" hits="1" branch="False" />
         <line number="11" hits="1" branch="False" />
         <line number="12" hits="1" branch="False" />
         <line number="13" hits="1" branch="False" />
      </lines>
   </method>
</methods>
   <lines>
      <line number="10" hits="1" branch="False" />
      <line number="11" hits="1" branch="False" />
      <line number="12" hits="1" branch="False" />
      <line number="13" hits="1" branch="False" />
</lines>
```

**Comment by rklec** [comment]
So has this fix been released?

**Comment by daveMueller** [comment]
I didn't really check which PR fixed this but all changes in the master branch at that time have been released in [3.1.0](https://github.com/coverlet-coverage/coverlet/releases/tag/v5.7.0).
